Core/clim-core/commands/tables.lisp

```diff
@@ -407,77 +393,76 @@ examine the type of the command menu item to see if it is
     (let ((item (apply #'make-menu-item string type value args)))
       (%add-menu-item table item after))))
 
-(defun remove-menu-item-from-command-table (command-table string
+(defun remove-menu-item-from-command-table (table string
                                             &key (errorp t))
   "Removes item from the `command-table'."
-  (let ((table (find-command-table command-table))
-        (item (find-menu-item string command-table :errorp nil)))
-    (with-slots (menu) table
-      (if (and errorp (not item))
-          (error 'command-not-present :command-table-name
-                 (command-table-designator-as-name table))
+  (setf table (find-command-table table))
+  (with-slots (menu) table
+    (if-let ((item (find-menu-item string table :errorp nil)))
+      ;; When a keystroke is still present leave the item be.
+      (if (null (command-menu-item-keystroke item))
           (setf menu (delete string menu
                              :key #'command-menu-item-name
-                             :test #'string-equal))))))
+                             :test #'string-equal))
+          (setf (command-menu-item-name item) nil))
+      (when errorp
+        (error 'command-not-present :command-table-name
+               (command-table-designator-as-name table))))))
 
 (defun map-over-command-table-keystrokes
     (function command-table &key (inherited t))
-  (declare (ignore inherited))
-  (let ((command-table (find-command-table command-table)))
-    (with-slots (keystroke-accelerators keystroke-items)
-        command-table
-      (loop for gesture in keystroke-accelerators
-            for item in keystroke-items
-            do (funcall function
-                        (command-menu-item-name item)
-                        gesture
-                        item)))))
-
-(defun find-keystroke-item (gesture command-table
+  (let ((table-object (find-command-table command-table)))
+    (flet ((map-table-entries (table)
+             (mapc #'(lambda (item)
+                       (with-slots (menu-name keystroke) item
+                         (when keystroke
+                           (funcall function menu-name keystroke item))))
+                   (slot-value table 'menu))))
+      (map-table-entries table-object)
```

Comment:
Why is this a local function? It is called immediately and only at one point.

Reply:
it is more readable to me that way -- a single task is encapsulated in a function, and then we call over inherited tables. Same applies to the second function -- do you want me to change that?

Reply:
I mean `map-table-entries` is only called once immediately after its definition. It does not take part in the recursive traversal as a functional argument or anything. In other words: why not replace the `flet` and the call with the body of `map-table-entries`?

Reply:
I have no issue with changing that -- my point is that I expect that the compiler will inline it anyway, but it is more readable (at least to me) when that part of the functionality is isolated as a (local) function.

Reply:
Please change it, if you don't mind. I'm not concerned with efficiency either, but abstracting this, in particular a single form, as function signals (to me) that it is going to be used more than once, so this is surprising. It is also longer and more complex than the direct version. A comment above the form could be good, though.